src/axiom_use.rs

```diff
@@ -1,9 +1,137 @@
 //! Generation of the `axiom_use` file.
+use std::sync::Arc;
+
+use itertools::PeekingNext;
+
 use crate::bit_set::Bitset;
+use crate::diag::Diagnostic;
+use crate::segment::SegmentRef;
+use crate::segment_set::SegmentSet;
+use crate::statement::{CommandToken, StatementAddress, TokenPtr};
 use crate::util::HashMap;
 use crate::StatementType;
 use crate::{as_str, database::time, Database};
 
+/// Diagnostics issued when checking axiom usage in the Database.
+///
+#[derive(Debug, Default)]
+pub struct UsageResult(Vec<(StatementAddress, Diagnostic)>);
+
+impl UsageResult {
+    /// Returns the list of errors that were generated during the usage check pass.
+    #[must_use]
+    pub fn diagnostics(&self) -> Vec<(StatementAddress, Diagnostic)> {
+        self.0.clone()
+    }
+}
+
+struct UsagePass<'a> {
+    axiom_use_map: HashMap<TokenPtr<'a>, Bitset>,
+    axioms: Vec<TokenPtr<'a>>,
+    result: &'a mut UsageResult,
+}
+
+impl<'a> UsagePass<'a> {
+    // Parses the 'usage' commmands in the database,
+    fn parse_command(
+        &mut self,
+        sref: SegmentRef<'_>,
+        args: &[CommandToken],
+    ) -> Result<(), Vec<Diagnostic>> {
+        use CommandToken::*;
+        let buf = &**sref.buffer;
+        match args {
+            [Keyword(cmd), label, Keyword(avoids), ..]
```

Comment:
```suggestion
            [Keyword(cmd), label, Keyword(avoids), rest @ ..]
```
and use it below instead of `&args[3..]`

Reply:
Thanks, done with last commit!